### ISSUE TITLE
`delete` operator not recognized as an assignment

### ISSUE THREAD

**Issue by weswigham** [Microsoft/TypeScript#8398]
**TypeScript Version:** 

nightly (1.9.0-dev.20160217)

**Code**

```ts
var z: { foo?: { bar: string; }; } = {};

z.foo = {bar: ''}

delete z.foo;

z.foo.bar; // Should be an error
```

**Expected behavior:**
Under strict-null mode, usages of `delete` are seen as, at least, assignments of `undefined`.

**Actual behavior:** 
Compiler doesn't detect `delete` as object-mutating.

**Comment by ahejlsberg** [comment]
Yup, we should fix that.
